lib/libzfs/libzfs_import.c

```diff
@@ -248,6 +248,17 @@ is_mpath_whole_disk(const char *path)
 	return (wholedisk);
 }
 
+static int
+udev_device_is_ready(struct udev_device *dev)
+{
+#ifdef HAVE_LIBUDEV_UDEV_DEVICE_GET_IS_INITIALIZED
+	return (udev_device_get_is_initialized(dev));
+#else
+	/* wait for DEVLINKS property to be initialized */
+	return (udev_device_get_property_value(dev, "DEVLINKS") != NULL);
```

Comment:
If the device is initialized, `udev_device_get_is_initialized()` is O(1) while `udev_list_entry_get_by_name()` is at least O(log(N)) as it does a couple binary searches in a sorted array. If it is not initialized, then it just looks in the database while `udev_device_get_property_value` will go to the uevent file before the database.

The optimization is probably not that useful given that we are running this in a loop with `usleep()`.